contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -0,0 +1,251 @@
+            _depositInfo.tokenAmount,
+            _depositInfo.grantor,
+            _roleData.grantee
+        );
+    }
+
+    function _unequipWearable(uint256 _gotchiId, uint256 _wearableToUnequip, uint256 _balanceToUnequip) internal {
+        uint256 _unequipedBalance;
+        for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
+            if (_unequipedBalance == _balanceToUnequip) break;
+            if (s.aavegotchis[_gotchiId].equippedWearables[slot] != _wearableToUnequip) continue;
+            s.aavegotchis[_gotchiId].equippedWearables[slot] = 0;
+            _unequipedBalance++;
+        }
+
+        LibItems.removeFromParent(address(this), _gotchiId, _wearableToUnequip, _unequipedBalance);
+        emit LibERC1155.TransferFromParent(address(this), _gotchiId, _wearableToUnequip, _unequipedBalance);
+    }
+
+    function withdraw(uint256 _depositId) public onlyOwnerOrApproved(s.itemsDeposits[_depositId].grantor, s.itemsDeposits[_depositId].tokenAddress) {
+        DepositInfo memory _depositInfo = s.itemsDeposits[_depositId];
+        require(
+            s.itemsRoleAssignments[_depositId].expirationDate < block.timestamp || s.itemsRoleAssignments[_depositId].revocable,
+            "ItemsRolesRegistryFacet: token has an active role"
+        );
+
+        delete s.itemsDeposits[_depositId];
+
+        _transferFrom(address(this), _depositInfo.grantor, _depositInfo.tokenAddress, _depositInfo.tokenId, _depositInfo.tokenAmount);
+
+        emit Withdrew(_depositId, _depositInfo.grantor, _depositInfo.tokenAddress, _depositInfo.tokenId, _depositInfo.tokenAmount);
+    }
+
+    function setRoleApprovalForAll(address _tokenAddress, address _operator, bool _isApproved) external override {
+        s.itemsTokenApprovals[msg.sender][_tokenAddress][_operator] = _isApproved;
+        emit RoleApprovalForAll(_tokenAddress, _operator, _isApproved);
+    }
+
+    /** View Functions **/
+
+    function roleData(
+        uint256 _depositId,
+        bytes32 _role,
+        address _grantee
+    ) external view override validRoleAndGrantee(_role, _grantee, _depositId) returns (RoleData memory) {
+        return s.itemsRoleAssignments[_depositId];
+    }
+
+    function roleExpirationDate(
+        uint256 _depositId,
+        bytes32 _role,
+        address _grantee
+    ) external view override validRoleAndGrantee(_role, _grantee, _depositId) returns (uint64 expirationDate_) {
+        return s.itemsRoleAssignments[_depositId].expirationDate;
+    }
+
+    function isRoleApprovedForAll(address _tokenAddress, address _grantor, address _operator) public view override returns (bool) {
+        return s.itemsTokenApprovals[_grantor][_tokenAddress][_operator];
+    }
+
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC1155Receiver, IERC165) returns (bool) {
+        return interfaceId == type(ISftRolesRegistry).interfaceId || interfaceId == type(IERC1155Receiver).interfaceId;
+    }
```

Comment:
Remove this function, it's probably already implemented somewhere in the Aavegotchi Diamond contract. We need to identify where is implemented, and simply add `ISftRolesRegistry`